### ISSUE TITLE
uninitialized constant Moped::BSON when using mongoid-observer gem

### ISSUE THREAD

**Issue by sdbondi** [digitalplaywright/mongoid-slug#194]
Due to the current method for checking for Mongoid 3 i.e checking for the Observer constant in Mongoid 
using mongoid-observer will break the check because it defines this constant.

Happy to make a PR, there are 2 ways I can think of to fix this:

Check the mongoid version 

```ruby
def self.mongoid3?
  unless defined?(@is_mongoid3)
     # Might need to check this part
     @is_mongoid3 = Gem::Version.new(Mongoid::VERSION)
          .between?(Gem::Version.new(3), Gem::Version.new(4))
  end
  @is_mongoid3 
end
```

Or perhaps check for `BSON` in `Moped`

```ruby
def self.mongoid3?
  ::Moped.const_defined?(:BSON)
end
```

Which has more to do with the Moped gem than Mongoid. Not sure if `mongoid3?` is used just to test for Moped::BSON exclusively. Perhaps it would be better to have `Mongoid::Slug.mod_bson` return either Moped::BSON or BSON and just use that method.

**Comment by dblock** [comment]
This is sloppy work on my part when implementing mongoid5 compat, I should have replaced all the calls to `mongoid3?`. Done in https://github.com/digitalplaywright/mongoid-slug/pull/195 with bringing in `mongoid-observers` and reproducing this error in tests.

**Comment by dblock** [comment]
I released 5.2.0 with this fix.

**Comment by sdbondi** [comment]
Awesome work! thanks! :+1:
